src/conventions/h-group/clue-finder/determine-clue.js

```diff
@@ -33,6 +33,10 @@ export function evaluate_clue(game, action, clue, target, target_card, bad_touch
 	logger.highlight('green', `------- ENTERING HYPO ${logClue(clue)} --------`);
 
 	const hypo_game = game.simulate_clue(action, { enableLogs: true });
+	hypo_game.last_actions = game.last_actions.slice();
+	hypo_game.last_actions[action.giver] = action;
+	hypo_game.handle_action({ type: 'turn', num: hypo_game.state.turn_count, currentPlayerIndex: hypo_game.state.nextPlayerIndex(hypo_game.state.ourPlayerIndex) }, true);
+	hypo_game.last_actions = game.last_actions;
```

Comment:
We only have to undo this because some tests try to convert the returned clues to JSON, and the clue contains the game state so if the game state contains a reference back to the clue it fails in tests as a circular structure.

I may need to restore the actionList which gets updated here:
https://github.com/WillFlame14/hanabi-bot/blob/553d061018027535804521eaf682d56283fec41c/src/action-handler.js#L23

Reply:
Done in latest commit.

Reply:
This seems unnecessary just for tests, I think it's better to modify extra-asserts.js so that this doesn't occur.

You could wrap the two asserts in try/catch, re-throw AssertionErrors and throw a different AssertionError upon catching a TypeError (for circular JSON)?

Reply:
I looked this up and inspect from node's util library seems like a simpler / better solution.